### ISSUE TITLE
Memory Card recognition with PS3 adapter

### ISSUE THREAD

**Issue by Mesandro** [ShendoXT/memcardrex#55]
hello, first of all congratulations for this fantastic software. I am using version 1.9.2b7a018 on windows using the original adapter for PS3 with WinUSB drivers (Zadig). I noticed that the software recognizes only the original Sony Memory Cards, while none of the other brands (I tried 4 different ones) are recognized. These however work if I use the adapter on PS3. I would like to understand if it is a limitation of the software or the drivers and how I can get around this. Thanks!

**Comment by ShendoXT** [comment]
Can you try with orion's ps1 save manager? It will use the same driver.
Does the same thing happen? http://onorisoft.free.fr/retro.htm?psx/psx.htm

**Comment by Mesandro** [comment]
Hi, thanks for the answer. I tried with the software you indicated, it behaves exactly the same way.. so I suppose it's the drivers

**Comment by ShendoXT** [comment]
What 3rd party Memory Cards are you using? Generic white 1 mega ones or something else?
And when you say that it's working on PS3 have you tried writing saves to them or just reading?

**Comment by Mesandro** [comment]
They are unofficial memory cards bought at the time of PlayStation (they were not purchased in modern times). So it's not the 1 mega white generics that are now found just about everywhere. On PlayStation 3 I only tried to read them and they are all read correctly

**Comment by ShendoXT** [comment]
Ok, I see. I'm trying to figure out if those cards require the 7V line.
Can you try writing to them on the PS3?

**Comment by Mesandro** [comment]
Hi, so I did some tests and I'm sharing them with you. First of all I have to apologize because I said something wrong. Since I moved old saves from these memory cards to PS3 a year ago, I was convinced that I had done it via an adapter directly to the console. Instead I remembered that I had first moved the files to original memory cards and then brought them to PS3 via an adapter. So the situation is different and I'll describe it to you in detail. There is still the problem that the adapter does not read memory cards from other brands, but at this point I suppose it's normal. I tried the same ones on PS1, PS2 and PS3 (adapter).

Original Sony Memory Cards:

PS1 - Read OK, Write OK
PS2 - Read OK, Write OK
PS3 Adapter - Read OK, Write OK

Other Brand Memory Cards (tried 4 different ones)

PS1 - Read OK, Write OK
PS2 - Read OK, Write OK
PS3 Adapter - Read NO, Write NO

In conclusion, the adapter behaves exactly like on the console via your software. Now I just have to wonder if it should also read non-original memory cards, but I don't know because I don't have any others to try. I hope these tests can be useful to you, and I apologize again for my initial incorrect report.

**Comment by ShendoXT** [comment]
The usual problem with 3rd party Memory Cards, especially older ones is that they use 5V flash memory instead
of the 3.3V one. Back in the day that memory was way cheaper and since 7V was available on the slot manufacturers just stepped those 7V down to 5V and called it a day.

Since Sony never used 7V for the official cards they stopped supplying 7V on the slot in the later slim PS2 models
and the accessories like this PS3 adapter. For slim PS2s you can solder a bridge wire from the controller port
but on the PS3 adapter you would need a step up converter from 5V to 7V. Or you can wire it directly to 5V, like I did with MemCARDuino, some cards work that way.

The reason some cards work with 5V on 7V line is again because of the cheapness of the manufacturers.
Some cards didn't even step down those 7V to 5V, rather they just ran the chips overvolted, leading to more heat, wear and in the end shorter lifespan. But those cards will work with 5V on that line since 5V is actually their flash rated spec.
Other cards which did use step down voltage regulator (or diode, gotta go cheap) will step 5V down to 3V and fail to work properly. Sometimes you can read and not write.

Anyway, I would always recommend 1st party Memory Cards or at least something that uses the proper 3.3V flash memory.

**Comment by Mesandro** [comment]
All very clear, thank you very much for your precious technical information. In my case, since I use memory cards for strictly personal use and for my passion for retrogaming, I will just need to use the original memory cards to not have any problems even with Memcardrex.

Small OT: So far I have always used the software in Windows, but as a Mac user I would like to try the preliminary version available. Are drivers also necessary on Mac?

**Comment by ShendoXT** [comment]
To be honest I haven't tried PS3 MCA on Mac myself but the interface is there and operational.
In theory you need to install libusb library on the Mac and it just should work.
All the intricate details are handled in software via MemcardRex.

I've had reports of people getting it to work on linux via mono with libusb.
